### ISSUE TITLE
Imported videos are 1 second less than the actual video being imported

### ISSUE THREAD

**Issue by McFlat** [Chocobozzz/PeerTube#1310]
I noticed that videos I've imported are always 1 second shorter than the actual video I've imported. Seems to me that if we transcode the video over and over again it will shave 1 second off of the video each time. Not sure what the fix would be, maybe we are missing a step in our process, like a validation step of the video length or the amount of frames in the video, etc. It seems to happen with every video I've imported from youtube so far.

Steps to reproduce are:

- Find a video on youtube and import it
- then notice how the time is 1 second less than the actual video on youtube

**Comment by Nutomic** [comment]
Something similar was already reported in #929. Does this only happen with videos from youtube, or also videos you upload directly? And can you check if the actual video file is shorter when you download it?

**Comment by McFlat** [comment]
@Nutomic I have only tested with videos imported from youtube so far, will have to test further with other sources and just with uploaded videos. I've been so busy with so much other stuff that import from youtube is all I've done so far, but I noticed that something looks wrong about it being 1 second off, so there's that. Once I have a chance to test with other sources and with uploading directly I'll make sure to mention it, until then this is all I got.
